utils/context_helpers.py

```
@@ -167,227 +167,284 @@ def get_all_summaries() -> str:
-            conversation_text += f"\n{role}: {msg['content']}"
 
-    # Add special section for image generation if we found any
-    if image_generation_results:
-        conversation_text += "\n\nIMAGE GENERATION RESULTS:\n" + "\n".join(
-            image_generation_results
+def _prioritize_and_truncate_content(file_contents: str, code_skeletons: str, images_info: str) -> Dict[str, str]:
+    """
+    Intelligently prioritize and truncate content based on importance.
+    Returns a dictionary with prioritized content sections.
+    """
+    MAX_TOTAL_CHARS = 150000  # Reduced from 200k for better performance
```

Ruff (0.12.2)

Local variable `MAX_TOTAL_CHARS` is assigned to but never used

Remove assignment to unused variable `MAX_TOTAL_CHARS`

(F841)